manager/controllers/app/blueprint_controller.go

```diff
@@ -310,11 +322,20 @@ func (r *BlueprintReconciler) reconcile(ctx context.Context, cfg *action.Configu
 	blueprint.Labels[managerUtils.BlueprintNameLabel] = blueprint.Name
 	blueprint.Labels[managerUtils.BlueprintNamespaceLabel] = blueprint.Namespace
 
-	for instanceName, module := range blueprint.Spec.Modules {
+	for instanceName := range blueprint.Spec.Modules {
+		module := blueprint.Spec.Modules[instanceName]
+		appDetails := fapp.ApplicationDetails{
+			Context:          blueprint.Spec.Application.Context,
+			WorkloadSelector: meta.LabelSelector{},
+		}
+		// pass workload selector only to the module that is exposed as a virtual endpoint to the workload
+		if module.Network.Endpoint {
+			appDetails.WorkloadSelector = blueprint.Spec.Application.WorkloadSelector
```

Comment:
Do we really need to pass to the module the workload selector? The application contacts the module, not the other way around. And if we do, this information is not enough.
I vote for removing WorkloadSelector from the helm values.

Reply:
removed